src/components/header/ProfileThreadJankOverview.js

```diff
@@ -4,24 +4,43 @@
 
 // @flow
 
+import React from 'react';
 import { connect } from 'react-redux';
 import IntervalMarkerOverview from './IntervalMarkerOverview';
+import MarkerTooltipContents from '../shared/MarkerTooltipContents';
 import { selectorsForThread } from '../../reducers/profile-view';
 import {
   styles,
   overlayFills,
 } from '../../profile-logic/interval-marker-styles';
-import { getSelectedThreadIndex } from '../../reducers/url-state';
+import {
+  getSelectedThreadIndex,
+  getImplementationFilter,
+} from '../../reducers/url-state';
 
 export default connect((state, props) => {
   const { threadIndex } = props;
   const selectors = selectorsForThread(threadIndex);
   const threadName = selectors.getFriendlyThreadName(state);
+  const thread = selectors.getThread(state);
   const selectedThread = getSelectedThreadIndex(state);
+  const implementationFilter = getImplementationFilter(state);
+  const getTooltipContents = item => {
+    return (
+      <MarkerTooltipContents
+        marker={item}
+        threadName={threadName}
+        thread={thread}
+        implementationFilter={implementationFilter}
+      />
+    );
+  };
```

Comment:
about the "HOC" I had in mind (actually not a HOC), I think it was something similar to the function you already have, except make it a functional stateless component:
```javascript
const TooltipContents = ({ item }: { string }) =>
  <MarkerTooltipContents
     marker={item}
     threadName={threadName}
     thread={thread}
     implementationFilter={implementationFilter}
  />;
```

So that in `IntervalMarkerOverview` you can use it like this:
```
render() {
  const { TooltipContents, ... } = this.props;
  return (
    stuff
    <TooltipContents item={XXX}/>
  );   
}
```
So this is very similar to what you have but it's maybe more "react". So I think this is better to do it like this :)

Yet It provides no performance benefit and I think the best idea is to memoize this. We already use `memoize-immutable` in this project so you can just use it here, no additional dependency.

Reply:
To use `memoize` you'll need to have a separate function that will generate this stateless component.

This separate function should take the `thread`, the `threadName` and the `implementationFilter`. (I mention it because I considered taking `state` and `threadIndex` but then the memoization wouldn't work when `state` changes). Then maybe you could even extract it and reuse it in both places already memoized after all :)

(note: memoize caches all calls by default, which I think is fine for us)

Reply:
at this point, maybe it's good to just make MarkerTooltipContents a connected component :)